packages/app/src/Util.test.ts

```diff
@@ -0,0 +1,40 @@
+import { splitByUrl } from "./Util";
+
+describe("splitByUrl", () => {
+  it("should split a string by URLs", () => {
+    const inputStr =
+      "@npub1q6mcr8t not https://example.com- sure what your stack is, https://example.com but I made a https://example.com! simple example (https://example.com) of how https://example.com/yo-yo https://example.example.com to do this https://example.com, https://example.com?q=asdf for Next.js apps hosted on Vercel https://example.com. Scarcity in money provides the incentive to create abundance in other things as there is a mechanism to reliably store value. https://i.imgur.com/rkqhjeq.png Every form of money that could be inflated by way of force or technological advancement has been.";
+    const expectedOutput = [
+      "@npub1q6mcr8t not ",
+      "https://example.com-",
```

Comment:
oops this isn't correct. the dash shouldn't match on a tld.

Reply:
maybe we don't need to worry about that case since i don't see a reason why anyone would type a dash directly after a URL without a space.

Reply:
Yea that would just be an invalid TLD